packages/product-react/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,220 @@
+
+// console.log({ stuff });
+
+import './Tabs.scss';
+
+interface ITabRefs {
+  [key: string]: React.RefObject<HTMLButtonElement>;
+}
+
+export interface ITab {
+  id: string;
+  ariaLabel: string;
+  label: string;
+}
+
+export interface IProps {
+  tabs: ITab[];
+  defaultActiveTab?: string;
+  onSelect?: (selectedTabId: string) => void;
+  compact?: boolean;
+}
+
+export const Tabs: React.FC<IProps> = props => {
+  const { tabs, onSelect, compact } = props;
+  const [firstTab] = tabs;
+  const defaultTab = props.defaultActiveTab || firstTab.id;
+  const [selectedTabId, setSelectedTabId] = React.useState<string>(defaultTab);
+  const [buttonsFocused, setButtonsFocused] = React.useState(false);
+  const onButtonFocus = () => setButtonsFocused(true);
+  const onButtonBlur = () => setButtonsFocused(false);
+
+  const setActiveTab = (id: string) => () => {
+    setSelectedTabId(id);
+
+    if (onSelect) {
+      onSelect(id);
+    }
+  };
+
+  const { tabRefs, focusedTab } = useFocusManager({
+    tabs,
+    firstFocusedId: defaultTab,
+    buttonsFocused
+  });
+
+  React.useEffect(() => {
+    setActiveTab(focusedTab)();
+  }, [focusedTab]);
+
+  if (tabs.length === 0) {
+    return null;
+  }
+
+  return (
+    <TabContext.Provider value={{ selectedTab: selectedTabId, tabs }}>
+      <div className="ray-tabs__wrapper" role="tablist">
+        {tabs.map(tab => {
+          const tabIsActive = tab.id === selectedTabId;
+
+          return (
+            <div
+              key={tab.id}
+              className={classnames('ray-tabs__button-wrapper', {
+                'ray-tabs__button-wrapper--active': tabIsActive
+              })}
+            >
+              <Button
+                type="tertiary"
+                aria-label={tab.ariaLabel}
+                aria-selected={tabIsActive}
+                onClick={setActiveTab(tab.id)}
+                role="tab"
+                aria-controls={tab.id}
+                ref={tabRefs[tab.id]}
+                tabIndex={tabIsActive ? 0 : -1}
+                data-testid="tab-button"
+                onFocus={onButtonFocus}
+                onBlur={onButtonBlur}
+                compact={compact}
+              >
+                {tab.label}
+              </Button>
+            </div>
+          );
+        })}
+      </div>
+      {props.children}
+    </TabContext.Provider>
+  );
+};
+
+Tabs.displayName = 'Tabs';
+
+export default Tabs;
+
+interface IFocusManager {
+  tabs: ITab[];
+  firstFocusedId: string;
+  buttonsFocused: boolean;
+}
+
+function useFocusManager(props: IFocusManager) {
+  const { tabs, firstFocusedId, buttonsFocused } = props;
+  const [focusedTab, setFocusedTab] = React.useState(firstFocusedId);
+  const [isMounted, setIsMounted] = React.useState(false);
+
+  const tabRefs: ITabRefs = tabs.reduce((acc, { id }) => {
+    return { ...acc, [id]: React.useRef<HTMLButtonElement>(null) };
+  }, {});
+
+  React.useEffect(() => {
+    if (!isMounted) {
```

Comment:
Is this used anywhere?

Reply:
Yeah, this is to make the hook listen to changes to `focusedTab` but ignore the first render (mount).